### PR TITLE
CDAP-5062 make explore jdbc fetch size configurable

### PR DESCRIPTION
**Pull request by albertshau**
Added a 'fetch.size' connection setting that controls how big the
fetch size should be. Also increasing the default fetch size from
50 to 1000 to make the default value more performant.

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-DUT3599

File: cdap-explore-jdbc/src/main/java/co/cask/cdap/explore/jdbc/ExploreStatement.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
 public class ExploreStatement implements Statement {
   private static final Logger LOG = LoggerFactory.getLogger(ExploreStatement.class);
 
-  private int fetchSize = 50;
+  private int fetchSize = 1000;
```
Comment (anwar6953) [inline cdap-explore-jdbc/src/main/java/co/cask/cdap/explore/jdbc/ExploreStatement.java]: rename it defaultFetchSize?

Comment (poornachandra) [inline cdap-explore-jdbc/src/main/java/co/cask/cdap/explore/jdbc/ExploreStatement.java]: The memory required is based on the size of an individual row times fetch size. So 1000 might be too big. Based on the row size, we should let the user increase it if required. What do you say?

Comment (albertshau) [inline cdap-explore-jdbc/src/main/java/co/cask/cdap/explore/jdbc/ExploreStatement.java]: this variable is not a default, but the actual, mutable fetch size for the statement. java.sql.Statement defines a setFetchSize() for people to set this.

50 is the Hive jdbc driver's default, which I guess is why we have this at 50.  But it's very slow when working with any decent number of rows.  It's also more likely to overload the cdap router.

**Comment by sreevatsanraman** [comment]
lgtm

File: cdap-explore-jdbc/src/main/java/co/cask/cdap/explore/jdbc/ExploreStatement.java (hunk 1 above)
Comment (anwar6953) [inline cdap-explore-jdbc/src/main/java/co/cask/cdap/explore/jdbc/ExploreStatement.java]: :+1: 
@poornachandra ?

Comment (poornachandra) [inline cdap-explore-jdbc/src/main/java/co/cask/cdap/explore/jdbc/ExploreStatement.java]: Even if we take each row to be on an average 100 KB, fetching 1000 rows at a time would need about 100 MB of memory for each fetch. So I think this is okay.

**Comment by albertshau** [comment]
added documentation about the new setting

**Comment by awholegunch** [comment]
docs LGTM.

**Pull request merged**